### DISCUSSION TITLE
Undeclared variables

### DISCUSSION THREAD

**Discussion by wanexa** [terraform-linters/tflint#1876] [Q&A]
Hello

was trying latest version and was wondering if tflint check variable name ? Like you declare it in tfvars file and use it in your code but with a letter missing or smth like that, it dosen't return an error
Is there a way to check this ?

**Comment by bendrucker** [comment]
This is a job for `terraform validate`!

**Reply by wanexa** [reply]
No it doesn't work , that's why I checked with tflint

**Reply by bendrucker** [reply]
Bad references are handled by `terraform validate`:

```tf
variable "foo" {}

locals {
  foo = var.fo
}
```

Always include examples in any issue/discussion. Sounds like you're referring to something like this:

```tf
# terraform.tfvars
fo = "bar"
```

Since `tfvars` files are not part of a Terraform module and are instead a way of passing options, neither `terraform validate` nor `tflint` consider them. 

Terraform emits errors from `plan` and `apply`. I'm gonna go ahead and transfer this back and open this as a planned enhancement. TFLint should have the same behavior. Meaning this should error:

```sh
tflint --var fo=bar
```
